### DISCUSSION TITLE
sum of table column

### DISCUSSION THREAD

**Discussion by samg666** [Budibase/budibase#3071] [Help] [answered]
Hello,

I have a data provider and table named Bills, I would like to put a paragraph to show the sum of the field named amount. On the bindable values, I can see a  value Bills.Rows, and I tried to bind to the paragraph and got `[object object],[object object],[object object]`. Can someone share some idea how to iterate the rows and got the field of amount in sum?

**Comment by asemplinski** [comment, marked as answer]
Disclaimer: I'm not sure if this is the most ideal/maintainable solution to this issue as I just started experimenting with budibase this weekend and am used to working with other enterprise low code platforms. AKA - take this with a grain of salt. 

Assuming you're using a table within the Budibase DB, the column is a number datatype and you want the sum of the column for that entire table and not what's shown on the UI. Given the way that the UI interacts with the backend and lack of UI aggregate functionality, it seems that the most straight forward way isn't to "iterate" over items but is to create a view for the Table that you want a sum for and ensure that it is only one record long. Then on the page that you'd like to display the data, drag a data provider where you need it and configure it to use the view you just created. 

Within the data provider, place a Repeater element. Finally, within the Repeater element place a Paragraph tag and choose the value from the Repeater element that corresponds to the sum value that you created. Due to the view only being one record, you should only see the value you calculated in the view.

**Reply by mjashanks** [reply]
Thanks Alex - I agree with this approach!

Also - given your experience in low-code, Id love to have a chat with you sometime to get some feedback. I'd really appreciate it - send me an email if you have some time for me 😀. My email is in my profile.

**Reply by samg666** [reply]
Thanks Alex, it seem a good approach. I have put a new data provide and share the same custom SQL parameters for sum/calculation purpose, then bind the value to a paragraph.
